staging/src/k8s.io/kubectl/pkg/cmd/cmd.go

```diff
@@ -465,6 +468,25 @@ func HandlePluginCommand(pluginHandler PluginHandler, cmdArgs []string) error {
 	return nil
 }
 
+// CommandHeaderRoundTripper adds a layer around the standard
+// round tripper to add Request headers before delegation.
+type CommandHeaderRoundTripper struct {
```

Comment:
I'd move this into its own package in `cli-runtime` so other folks can use it

Reply:
I've put it in the `genericclioptions` package of `cli-runtime`, which has the `ConfigFlags`. Is this OK, or do you want to create a separate package altogether in this repo?